src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,349 @@
+        }
+        return sw.toString();
+    }
+
+    /**
+     * Encodes the current stored information in PEM formated {@link File}
+     * 
+     * @throws IOException launched if a problem exists generating the PEM information or writing the {@link File}
+     */
+    public void encodePEM(@Nonnull File pemFile) throws IOException {
+        FileUtils.writeStringToFile(pemFile, encodePEM());
+    }
+
+    /**
+     * Obtain {@link KeyPair} object with the public and private key from the read PEM. No conversion is performed, the
+     * read PEM must contain private and public key in order to obtain a {@link KeyPair} object, {@link null} will be
+     * returned in all the other cases.
+     * 
+     * @return {@link KeyPair} object with public and private keys or {@link null} if the read PEM didn't contain
+     * private and public keys.
+     */
+    @Nullable
+    public KeyPair getKeyPair() {
+        if (pemObject instanceof KeyPair) {
+            return (KeyPair) pemObject;
+        } // We will need conversion here on BC 1.54
+        return null;
+    }
+
+    /**
+     * Obtain {@link PublicKey} object from the read PEM. If the PEM data contained other object type like
+     * {@link KeyPair} or {@link Certificate}, the public key will be extracted from them.
+     * 
+     * @return {@link PublicKey} with the public key, null if a public key could not be obtained from the current data
+     */
+    @Nullable
+    public PublicKey getPublicKey() {
+        if (pemObject instanceof PublicKey) {
+            return (PublicKey) pemObject;
+        } else if (pemObject instanceof KeyPair) {
+            return ((KeyPair) pemObject).getPublic();
+        } else if (pemObject instanceof Certificate) {
+            return ((Certificate) pemObject).getPublicKey();
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link Certificate} object from the read PEM.
+     * 
+     * @return {@link Certificate} with the certificate, null if a certificate could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public Certificate getCertificate() {
+        if (pemObject instanceof Certificate) {
+            return ((Certificate) pemObject);
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link PrivateKey} object from the read PEM. If the PEM data contained other object type like
+     * {@link KeyPair}, the private key will be extracted from them.
+     * 
+     * @return {@link PrivateKey} with the private key, null if a private key could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public PrivateKey getPrivateKey() {
+        if (pemObject instanceof PrivateKey) {
+            return (PrivateKey) pemObject;
+        } else if (pemObject instanceof KeyPair) {
+            return ((KeyPair) pemObject).getPrivate();
+        }
+        return null;
+    }
+    
+    /**
+     * Obtains raw JCA or BouncyCastle {@link Object} from the read PEM. 
```

Comment:
Maybe a documented list of known return values is worth here, and a reminder that it depends on the nature of the PEM source.